src/main/java/eu/europeana/metis/sandbox/controller/DatasetController.java

```diff
@@ -217,6 +237,19 @@ public ProgressInfoDto retrieveDataset(
         .collect(Collectors.toList());
   }
 
+  private InputStream createXsltAsInputStreamIfPresent(MultipartFile xslt) {
+    if(xslt != null && !xslt.isEmpty()){
+      checkArgument(xslt.getContentType().equals("application/xslt+xml"),
+          "The given xslt file should be a single xml file.");
+      try {
+        return new ByteArrayInputStream(xslt.getBytes());
+      } catch (IOException e) {
+        throw new XsltProcessingException("Something wrong happened while processing xslt file.", e);
+      }
+    }
+    return null;
```

Comment:
Perhaps ByteArrayInputStream(new byte[0])?

Reply:
That is definitely better, will change that